jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -205,6 +205,19 @@ var module = module;
         // if the macaroon is valid.
         false
       );
+    },
+
+    /**
+      Lists all the templates for cloud credentials for a specific user.
+
+      @method listTemplates
+      @param callback {Function} A callback to handle errors or accept the data
+          from the request. Must accept an error message or null as its first
+          parameter and the response data as its second.
+    */
+    listTemplates: function(callback) {
+      var url = this.jemURL + '/template';
+      return _makeRequest(this.bakery, url, 'GET', null, callback, true);
```

Comment:
It would be nice to not just propagate the data sent over the wire.
Transforming that data to some internal data structure ensures the client results are JS friendly and sane, we can easily switch API versions without potentially change internal code and also documents what's passed to callbacks, which is also something we need to add here, as reading this function does not inform me about what's data is actually passed to callbacks.